dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -304,11 +304,15 @@ def __init__(
                 # here we should set if to False again
                 self._ever_consumed = False
             except StopIteration:
-                assert False, (
-                    "It seems that there is no data in the pipeline. This may happen "
-                    "if `last_batch_policy` is set to PARTIAL and the requested batch size is "
-                    "greater than the shard size."
-                )
+                # This case might not be an error if we're iterating over pipeline that is
```

Comment:
how about putting this repeated logic into a function, or at least saving the error message to a variable?

Reply:
Good idea, thank you! Extracted to a function